pkg/js/libs/kerberos/kerberos.go

```diff
@@ -143,3 +143,49 @@ func asRepToHashcat(asrep messages.ASRep) (string, error) {
 		hex.EncodeToString(asrep.EncPart.Cipher[:16]),
 		hex.EncodeToString(asrep.EncPart.Cipher[16:])), nil
 }
+
+type TGS struct {
+	Ticket messages.Ticket
+	Hash   string
+}
+
+func (c *KerberosClient) GetServiceTicket(domain, controller string, username, password string, target, spn string) (TGS, error) {
+	var tgs TGS
+
+	if !protocolstate.IsHostAllowed(domain) {
+		// host is not valid according to network policy
+		return tgs, protocolstate.ErrHostDenied.Msgf(domain)
+	}
+
+	opts, err := newKerbrosEnumUserOpts(domain, controller)
+	if err != nil {
+		return tgs, err
+	}
+	cl := kclient.NewWithPassword(username, opts.realm, password, opts.config, kclient.DisablePAFXFAST(true))
```

Comment:
I notice `kclient` is never [destroy](https://pkg.go.dev/github.com/ropnop/gokrb5/v8@v8.0.0-20201111231119-729746023c02/client#Client.Destroy)ed, is this intentional?

Reply:
It is not intentional, I assumed there was a reason, as the `EnumerateUser` function does not destroy the client.

https://github.com/projectdiscovery/nuclei/blob/984dfaf49ae5cdb06c70828c72e4a9f7f9a7151c/pkg/js/libs/kerberos/kerberos.go#L86-L136

Should I add the relevant change? If so, should I add the `Destroy()` to `EnumerateUser` too or that's a different PR?

Reply:
Yeah, I think it would be best practice to do that. I'm sure the resources will get cleaned up eventually but it would be best not to leave them dangling.